src/_langs/en/updates/2015-07-22-access-ble-devices-with-web-bluetooth.markdown

```diff
@@ -0,0 +1,279 @@
+
+{% highlight javascript %}
+navigator.bluetooth.requestDevice({ filters: [{ services: ['battery_service'] }] })
+.then(device => device.connectGATT())
+.then(server => {
+  // Getting Battery Service...
+  return server.getPrimaryService('battery_service');
+})
+.then(service => {
+  if (!service) throw 'Battery Service not found';
+  // Getting Battery Level Characteristic...
+  return service.getCharacteristic('battery_level');
+})
+.then(characteristic => {
+  // Reading Battery Level...
+  return characteristic.readValue();
+})
+.then(buffer => {
+  var data = new DataView(buffer);
+  console.log('Battery percentage is ' + data.getUint8(0));
+})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+If you use a custom Bluetooth GATT characteristic,
+`window.BluetoothUUID.getCharacteristic` definitely comes in handy. Calling it
+with `0x5678` will return `"00005678-0000-1000-8000-00805f9b34fb"`.
+
+{% highlight javascript %}
+...
+.then(service => {
+  var myCharacteristic = BluetoothUUID.getCharacteristic(0x5678);
+  return service.getCharacteristic(myCharacteristic);
+})
+{% endhighlight %}
+
+### Write to a Bluetooth Characteristic
+
+Writing to a Bluetooth GATT Characteristic is as easy as reading it. This time,
+let's reset the value of the Energy Expended field in the Heart Rate
+Measurement characteristic to 0 on a heart rate monitor device.
+
+There is no magic there I promise. It's all explained in the [Heart Rate
+Control Point Characteristic
+page](https://developer.bluetooth.org/gatt/characteristics/Pages/CharacteristicViewer.aspx?u=org.bluetooth.characteristic.heart_rate_control_point.xml).
+
+{% highlight javascript %}
+navigator.bluetooth.requestDevice({ filters: [{ services: ['heart_rate'] }] })
+.then(device => device.connectGATT())
+.then(server => server.getPrimaryService('heart_rate'))
+.then(service => service.getCharacteristic('heart_rate_control_point'))
+.then(characteristic => {
+  var resetEnergyExpended = new Uint8Array([1]);
+  return characteristic.writeValue(resetEnergyExpended);
+})
+.then(function(value) {
+  console.log('Energy expended has been reset.');
+})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+## Samples
+
+Samples below have been tested on Chrome OS M45 with the Web Bluetooth flag
+enabled. To enjoy these samples to their fullest, I recommend you install the [BLE Peripheral Simulator Android
+App](https://play.google.com/store/apps/details?id=io.github.webbluetoothcg.bletestperipheral)
+which simulates a BLE Peripheral with a Battery Service or a Heart Rate
+Service.
+
+- Battery Level Sample - [https://googlechrome.github.io/samples/web-bluetooth](https://googlechrome.github.io/samples/web-bluetooth)
+- Reset Energy Expended Sample - [https://scheib.github.io/bluetooth-demos/heart-rate-write.html](https://scheib.github.io/bluetooth-demos/heart-rate-write.html)
```

Comment:
@g-ortuno and I don't think these samples are in shape to share. They're definitely not best practice code right now. We think that if we want to use these samples they should get cleaned up more and just folded into googlechrome.../samples.